process/io/plot_proc.py

```diff
@@ -13352,134 +13337,34 @@ def main(args=None):
-    page7 = plt.figure(figsize=(12, 9), dpi=80)
-    page8 = plt.figure(figsize=(12, 9), dpi=80)
-    page9 = plt.figure(figsize=(12, 9), dpi=80)
-    page10 = plt.figure(figsize=(12, 9), dpi=80)
-    page11 = plt.figure(figsize=(12, 9), dpi=80)
-    page12 = plt.figure(figsize=(12, 9), dpi=80)
-    page13 = plt.figure(figsize=(12, 9), dpi=80)
-    page14 = plt.figure(figsize=(12, 9), dpi=80)
-    page15 = plt.figure(figsize=(12, 9), dpi=80)
-    page16 = plt.figure(figsize=(12, 9), dpi=80)
-    page17 = plt.figure(figsize=(12, 9), dpi=80)
-    page18 = plt.figure(figsize=(12, 9), dpi=80)
-    page19 = plt.figure(figsize=(12, 9), dpi=80)
-    page20 = plt.figure(figsize=(12, 9), dpi=80)
-    page21 = plt.figure(figsize=(12, 9), dpi=80)
-    page22 = plt.figure(figsize=(12, 9), dpi=80)
-    page23 = plt.figure(figsize=(12, 9), dpi=80)
-    page24 = plt.figure(figsize=(12, 9), dpi=80)
-    page25 = plt.figure(figsize=(12, 9), dpi=80)
-    page26 = plt.figure(figsize=(12, 9), dpi=80)
-    page27 = plt.figure(figsize=(12, 9), dpi=80)
+    # Increase range when adding new page
+    pages = [plt.figure(figsize=(12, 9), dpi=80) for i in range(28)]
 
     # run main_plot
     main_plot(
-        page0,
-        page1,
-        page2,
-        page3,
-        page4,
-        page5,
-        page6,
-        page7,
-        page8,
-        page9,
-        page10,
-        page11,
-        page12,
-        page13,
-        page14,
-        page15,
-        page16,
-        page17,
-        page18,
-        page19,
-        page20,
-        page21,
-        page22,
-        page23,
-        page24,
-        page25,
-        page26,
-        page27,
+        pages,
         m_file,
         scan=scan,
         demo_ranges=demo_ranges,
         colour_scheme=colour_scheme,
     )
 
-    # with bpdf.PdfPages(args.o) as pdf:
-    with bpdf.PdfPages(args.f + "SUMMARY.pdf") as pdf:
-        pdf.savefig(page0)
-        pdf.savefig(page1)
-        pdf.savefig(page2)
-        pdf.savefig(page3)
-        pdf.savefig(page4)
-        pdf.savefig(page5)
-        pdf.savefig(page6)
-        pdf.savefig(page7)
-        pdf.savefig(page8)
-        pdf.savefig(page9)
-        pdf.savefig(page10)
-        pdf.savefig(page11)
-        pdf.savefig(page12)
-        pdf.savefig(page13)
-        pdf.savefig(page14)
-        pdf.savefig(page15)
-        pdf.savefig(page16)
-        pdf.savefig(page17)
-        pdf.savefig(page18)
-        pdf.savefig(page19)
-        pdf.savefig(page20)
-        pdf.savefig(page21)
-        pdf.savefig(page22)
-        pdf.savefig(page23)
-        pdf.savefig(page24)
-        pdf.savefig(page25)
-        pdf.savefig(page26)
-        pdf.savefig(page27)
+    if args.output_format == "pdf":
+        with bpdf.PdfPages(args.f + "SUMMARY.pdf") as pdf:
+            for p in pages:
+                pdf.savefig(p)
+    else:
+        folder_name = pathlib.Path(args.f.removesuffix(".DAT") + "_SUMMARY")
+        folder = pathlib.Path(folder_name)
```

Comment:
Second line doesnt do anything
```suggestion
        folder = pathlib.Path(args.f.removesuffix(".DAT") + "_SUMMARY")
```